pilot/model/validation.go

```diff
@@ -331,6 +345,9 @@ func ValidateMatchCondition(mc *proxyconfig.MatchCondition) (errs error) {
 
 // ValidateHTTPHeaderName checks that the name is lower-case
 func ValidateHTTPHeaderName(name string) error {
+	if name == "" {
+		return fmt.Errorf("header name cannot be empty")
```

Comment:
Minor nit: we could hoist this error out to be
```go
var errEmptyHeaderName = errors.New("header name cannot be empty")

...
      if name == "" {
           return errEmptyHeaderName
      }
```

Reply:
I agree this is good practice, but unfortunately it has not been followed anywhere else yet.  With the combined repo now for auth/broker/mixer/pilot this would now be exceptionally useful.  @odeke-em I recommend creating an issue for this and we can resolve in a separate PR